### PR TITLE
fix(components): normalize collection search text around parentheses

### PR DESCRIPTION
**Pull request by adriangohjw**
### 📊 PR diff breakdown

| Bucket | Files | +Lines | -Lines |
|---|---|---|---|
| ⚙️ App | 2 | +33 | -3 |
| 🧪 Test | 2 | +261 | -0 |
| 📄 Doc | 0 | +0 | -0 |
| 🚫 Ignore | 0 | +0 | -0 |

## Problem

Collection page search uses substring matching on item titles and descriptions. Titles pasted from PDFs or Word often contain Unicode characters that look identical on screen — fullwidth parentheses, non-breaking spaces, soft hyphens — but fail a raw `String.includes()` check. Users could search `MANAGEMENT` and see results, then lose them as soon as they typed `(FM)`.

## Solution

Normalize both the stored text and the user's search input before comparing: NFKC Unicode normalization, strip invisible characters, collapse whitespace, and treat optional spacing around parentheses as equivalent (e.g. `MANAGEMENT (FM)` matches `MANAGEMENT(FM)`).

**Breaking Changes**

- [ ] Yes - this PR contains breaking changes
  - Details ...
- [x] No - this PR is backwards compatible

**Features**:

- N/A

**Improvements**:

- N/A

**Bug Fixes**:

- Collection search now matches titles with fullwidth parentheses, non-breaking spaces, soft hyphens, and inconsistent spacing before parentheses.
- Search safely handles items with a missing description.

## Before & After Screenshots

**BEFORE**:

N/A — behaviour fix with no visual change when search already worked.

**AFTER**:

Screenshot 2026-08-03 at 3 47 27 PM

N/A

## Tests

`pnpm exec vitest run src/templates/next/layouts/Collection/utils/__tests__/getFilteredItems.test.ts src/templates/next/layouts/Collection/utils/__tests__/normalizeCollectionSearchText.test.ts` (from `packages/components`)

**Manual Verification Steps**:

- [ ] Open a published site with a collection page that has items whose titles include parentheses (e.g. `(FM)`).
- [ ] Type a full title fragment including `(FM)` into the collection search bar and confirm matching articles appear.
- [ ] Type a partial fragment such as `management (FM)` and confirm items like `Facilities Management (FM) …` still match.
- [ ] Clear the search and confirm all articles return.

**New scripts**:

- N/A

**New dependencies**:

- N/A

**New dev dependencies**:

- N/A

<div><a href="https://cursor.com/agents/bc-72496684-ef5d-4711-bca6-af41b1fc476d?cursor_ref=pr_footer&cursor_cta=open_in_web"><picture><source media="(prefers-color-scheme: dark)" srcset="https://cursor.com/assets/images/open-in-web-dark.png"><source media="(prefers-color-scheme: light)" srcset="https://cursor.com/assets/images/open-in-web-light.png">Open in Web</picture></a>&nbsp;<a href="https://cursor.com/background-agent?bcId=bc-72496684-ef5d-4711-bca6-af41b1fc476d&cursor_ref=pr_footer&cursor_cta=open_in_cursor"><picture><source media="(prefers-color-scheme: dark)" srcset="https://cursor.com/assets/images/open-in-cursor-dark.png"><source media="(prefers-color-scheme: light)" srcset="https://cursor.com/assets/images/open-in-cursor-light.png">Open in Cursor</picture></a>&nbsp;</div>

<details><summary><h3>Greptile Summary</h3></summary>

Improves Collection search matching for text copied from sources that use compatibility Unicode characters, invisible formatting characters, Unicode whitespace, or different spacing around parentheses.

Collection filtering was exercised with the focused test suite and review-only exact-path checks. Whitespace- and format-only queries retained all items, while visually equivalent title and description text matched across the tested Unicode and spacing variations.
</details>

<details><summary><h3>Confidence Score: 5/5</h3></summary>

</details>

<details><summary><h3><a href="https://www.greptile.com/trex">T-Rex</a> T-Rex Logs</h3></summary>

**What T-Rex did**
- Ran the focused collection filtering baseline tests for getFilteredItems, and all 12 tests passed.
- Executed a review-authored Vitest test that invokes getFilteredItems with whitespace and Unicode variants; all 3 checks passed.
- Confirmed the production filtering path retains items for empty normalized input and matches tested Unicode and parenthesis-spacing variants.
- Cross-validated Unicode edge-case flow by reviewing the baseline and after-test outputs, and the edge-case suite passed.

<a href="https://app.greptile.com/trex/runs/17024443/artifacts"><picture><source media="(prefers-color-scheme: dark)" srcset="https://greptile-static-assets.s3.amazonaws.com/badges/ViewAllArtifactsDark.svg?v=4"><source media="(prefers-color-scheme: light)" srcset="https://greptile-static-assets.s3.amazonaws.com/badges/ViewAllArtifacts.svg?v=4">View all artifacts</picture></a>

<sub><a href="https://www.greptile.com/trex">T-Rex</a> Ran code and verified through T-Rex</sub>
</details>

<sub>Reviews (1): Last reviewed commit: ["refactor(components): drop matchesCollec..."](https://github.com/opengovsg/isomer/commit/4cb0dccc34ef6044763a4d3a60deb4698c13da69) | [Re-trigger Greptile](https://app.greptile.com/api/retrigger?id=49598758)</sub>

### CODE REVIEW

File: packages/components/src/templates/next/layouts/Collection/utils/getFilteredItems.ts [hunk 1]
```diff
@@ -2,18 +2,26 @@ import type { ProcessedCollectionCardProps } from "~/interfaces"
...
 import { FILTER_ID_YEAR, NO_SPECIFIED_YEAR_FILTER_ID } from "./constants"
+import { normalizeCollectionSearchText } from "./normalizeCollectionSearchText"
 
 export const getFilteredItems = (
   items: ProcessedCollectionCardProps[],
   appliedFilters: AppliedFilter[],
   searchValue: string,
 ): ProcessedCollectionCardProps[] => {
+  const normalizedSearchValue =
+    searchValue !== "" ? normalizeCollectionSearchText(searchValue) : ""
```
Comment (adriangohjw) [inline packages/components/src/templates/next/layouts/Collection/utils/getFilteredItems.ts:13]: note: moved out as a minor optimization so we don't have to keep normalising the search value twice - we do it once now here

File: packages/components/src/templates/next/layouts/Collection/utils/normalizeCollectionSearchText.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
+export const normalizeCollectionSearchText = (text: string): string => {
+  return (
+    text
+      // NFKC: map compatibility forms to their usual ASCII equivalents (e.g. fullwidth （FM） → (FM)).
+      .normalize("NFKC")
+      // Format characters (ZWSP, soft hyphens, BOM, etc.) pasted from PDFs/Word.
+      .replace(/\p{Cf}/gu, "")
+      // All Unicode whitespace (NBSP, ideographic space, etc.) → regular spaces.
+      .replace(/\p{White_Space}+/gu, " ")
```
Comment (dcshzj) [inline packages/components/src/templates/next/layouts/Collection/utils/normalizeCollectionSearchText.ts]: `\p{Cf}` / `\p{White_Space}` property escapes are not supported until Firefox 78, but this package's browserslist declares `firefox >= 67`. The build is plain `tsc` (no regex downleveling), so this ships as-is: Firefox 67-77 will throw a `SyntaxError` parsing this chunk, breaking Collection search/filtering entirely (not a graceful degrade, since it fails before `isSupportedBrowser` even runs).

```suggestion
      // Format characters (ZWSP, soft hyphens, BOM, etc.) pasted from PDFs/Word.
      .replace(/[\u00AD\u200B-\u200F\u202A-\u202E\u2060-\u2064\uFEFF]/g, "")
      // All Unicode whitespace (NBSP, ideographic space, etc.) collapses to a single space; `\s` already matches this exact Unicode White_Space set per spec, no /u flag needed.
      .replace(/\s+/g, " ")
```

Comment (adriangohjw) [reply packages/components/src/templates/next/layouts/Collection/utils/normalizeCollectionSearchText.ts]: @cursoragent verify if theres any edge cases we missed out due to our supported browsers list, and update this section accordingly. also, update tests accordingly too

**Comment by adriangohjw** [comment]
@cursor fix failing CI

**Review by dcshzj — APPROVED** [review]
